### PR TITLE
feat(cache): expand ped

### PR DESCRIPTION
**Pull request by Aure7138**
The current two caching methods cannot read componentpeds_a_c.rpf, this pr adds a third caching method - cache on init and expand the number of peds on this basis

### CODE REVIEW

**Comment by Aure7138** [comment]
Currently finished adding a third caching method, everything is fine, the next step is to expand ped

image
image

If anyone has any suggestions please let me know

**Comment by xiaoxiao921** [comment]
long time since I wrote this but isnt hooking GetDevice better when injecting before game load?

**Comment by xiaoxiao921** [comment]
The whole point is that you don't have to use any yim packfile related functions and just utilize the fact that game is already reading all the files from the rpf as the game load, then you just profit off that

oh yeah another issue: the game doesnt load all dlc rpfs when loading in SP, meaning that they have to go to MP directly

**Comment by Aure7138** [comment]
> long time since I wrote this but isnt hooking GetDevice better when injecting before game load?

will try that one later

> The whole point is that you don't have to use any yim packfile related functions and just utilize the fact that game is already reading all the files from the rpf as the game load, then you just profit off that
> 
> oh yeah another issue: the game doesnt load all dlc rpfs when loading in SP, meaning that they have to go to MP directly

Ah, I see, the hook currently still need yim packfile, to do this probably need hook other functions, I'll try
By the way, I tried SP and MP, the amount seems to be the same, because the hook is special?

image
Get 1019 peds, need a method to get ped type from CPedModelInfo

File: src/services/gta_data/gta_data_service.cpp [hunk 1]
```diff
@@ -459,10 +456,16 @@ namespace big
 			}
 			for (auto it = peds.begin(); it != peds.end();)
 			{
-				if (!model_info::does_model_exist(it->m_hash))
-					peds.erase(it);
-				else
+				if (CPedModelInfo* info = model_info::get_model<CPedModelInfo*>(it->m_hash))
+				{
+					static std::array<std::string, 30> ped_types = {"PLAYER_0", "PLAYER_1", "NETWORK_PLAYER", "PLAYER_2", "CIVMALE", "CIVFEMALE", "COP", "GANG_ALBANIAN", "GANG_BIKER_1", "GANG_BIKER_2", "GANG_BIKER_2", "GANG_RUSSIAN", "GANG_RUSSIAN_2", "GANG_RUSSIAN_2", "GANG_JAMAICAN", "GANG_AFRICAN_AMERICAN", "GANG_KOREAN", "GANG_CHINESE_JAPANESE", "GANG_PUERTO_RICAN", "DEALER", "MEDIC", "FIREMAN", "CRIMINAL", "BUM", "PROSTITUTE", "SPECIAL", "MISSION", "SWAT", "ANIMAL", "ARMY"};
+					std::strncpy(it->m_ped_type, ped_types[*(uint32_t*)((uintptr_t)info + 0x138)].c_str(), sizeof(it->m_ped_type));
```
Comment (Yimura) [inline src/services/gta_data/gta_data_service.cpp]: Please don't use a static offset, modify the class on GTAV-Classes repo.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**